### DISCUSSION TITLE
It seems that pyg is difficult to deal with directed heterogeneous graphs？

### DISCUSSION THREAD

**Discussion by CocoGzh** [pyg-team/pytorch_geometric#3682] [Q&A]
I load the metapath2vec data according to the examples (https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html) and create my own data set. I found that transforms.ToUndirected() must be used to make the model work. 
I wonder if there is a solution to deal with directed heterogeneous data? For example, I just want to pass the message from "author" to "paper" in metapath2vec dataset. What should I do?
Now I can only create a GNN for each directed graph and manually fuse the intermediate results.

**Comment by rusty1s** [comment]
Can you show me a small example regarding your issue? PyG should be able to deal with directed graphs just fine. However, importantly, node features of types will not get updated when no edge type is pointing to it.
